### PR TITLE
Rename SeqMem to SyncReadMem - fixes #102.

### LINKED ISSUE #102: Call it SyncMem
Rename the new memory from sequential to synchronous memory (e.g., SyncMem). That's the term that is used for memory chips. E.g., SSRAM is a synchronous static RAM, in SDRAM the S is for synchronous.

Basically there are two types of memories (or better memory interfaces) available: asynchronous and synchronous. The term sequential is not use for memories. And if your really want to, I would say both asynchronous and synchronous are 'sequential'.

### PR DESCRIPTION
**Pull request by ucbjrl**
Retain un-deprecated SeqMem in compatibility mode, deprecate in chisel3.
Fixes #102.

### CODE REVIEW

File: chiselFrontend/src/main/scala/chisel3/core/Mem.scala [hunk 1]
```diff
@@ -143,7 +143,7 @@ object SeqMem {
   * @note when multiple conflicting writes are performed on a Mem element, the
   * result is undefined (unlike Vec, where the last assignment wins)
   */
-sealed class SeqMem[T <: Data](t: T, n: Int) extends MemBase[T](t, n) {
+sealed class SyncReadMem[T <: Data](t: T, n: Int) extends MemBase[T](t, n) {
   def read(addr: UInt, enable: Bool): T = {
```
Comment (jackkoenig) [inline chiselFrontend/src/main/scala/chisel3/core/Mem.scala]: Is this not a backwards incompatible change in the compatibility layer as well? What if someone matches on the type of a Mem? Obviously they will get a compiler error but yeah.

**Comment by ducky64** [comment]
"Does rocket still build with this?"

File: chiselFrontend/src/main/scala/chisel3/core/Mem.scala (hunk 1 above)
Comment (ucbjrl) [inline chiselFrontend/src/main/scala/chisel3/core/Mem.scala]: Don't these changes accomplish that (from src/main/scala/chisel3/compatibility.scala):
```
  val SeqMem = chisel3.core.SyncReadMem
  type SeqMem[T <: Data] = chisel3.core.SyncReadMem[T]
```

Comment (jackkoenig) [inline chiselFrontend/src/main/scala/chisel3/core/Mem.scala]: It does!
```scala
scala> class Foo(val a: Int)
defined class Foo

scala> type Bar = Foo
defined type alias Bar

scala> val bar = new Bar(3)
bar: Foo = Foo@33433fc5

scala> bar.isInstanceOf[Bar]
res0: Boolean = true

scala> bar match {
     | case _: Bar => print("Success!")
     | }
Success!
```
It doesn't work if you try to do reflection though:
```scala
scala> bar.getClass
res5: Class[_ <: Foo] = class Foo
```
But this is probably okay

**Comment by ucbjrl** [comment]
rocket-chip builds and passes [jenkins build 1116](http://a17.millennium.berkeley.edu/jenkins/job/rocket-chip_with_chisel3/1116/)

**Pull request merged**